src/validators/mod.rs

```diff
@@ -441,9 +442,12 @@ pub fn build_validator<'a>(
 /// but that would confuse it with context as per pydantic/pydantic#1549
 #[derive(Debug, Default)]
 pub struct Extra<'a> {
-    /// This is used as the `data` kwargs to validator functions, it also represents the current model
-    /// data when validating assignment
+    /// Represents the fields of the model we are currently validating
+    /// If there is no model this will be None
     pub data: Option<&'a PyDict>,
+    /// Represents the fields of the model we are currently validating
+    /// If there is no model this will be None
+    pub field_name: Option<&'a str>,
     /// The field being assigned to when validating assignment
     pub field: Option<&'a str>,
```

Comment:
It seems that `field` is being used to mean something else (something to do with validating assignments). I am a bit concerned that:
- `field` should have a more descriptive name, or we should rework other logic to allow this to function as the "field name of the last seen model field"
- `Extra` is becoming a dumping ground for things.
- The logic of triggering "completely different logic" based on something set in `extra` is a bit confusing at first glance

Reply:
extra being a dumping ground was kind of the intention to avoid having to change the signature of 40+ functions when adding an argument.

But I have no idea if using a struct instead of more arguments has a performance impact.

I'd be totally happy to all these things being arguments or another route.

Let's talk in a bit.